src/azul/service/hca_response_v5.py

```diff
@@ -298,9 +308,53 @@ def make_projects(self, entry):
                 translated_project['insdcProjectAccessions'] = project.get('insdc_project_accessions', [None])
                 translated_project['insdcStudyAccessions'] = project.get('insdc_study_accessions', [None])
                 translated_project['supplementaryLinks'] = project.get('supplementary_links', [None])
+                translated_project['contributorMatrices'] = self.make_contributor_matrices_tree(entry)
             projects.append(translated_project)
         return projects
 
+    def make_contributor_matrices_tree(self, entry) -> JSON:
+        """
+        Returns a stratification tree for the Contributor-generated matrix files
+        in the entry.
+        """
+        contributor_matrices = entry['contents']['contributor_matrices']
+        if contributor_matrices:
+            files = one(contributor_matrices)['file']
+        else:
+            return {}
+        unique_values = defaultdict(set)  # Unique values per stratification key
+        for file in files:
+            # Transform the stratification string into a list of dicts (1 per line)
+            file['stratification'] = [
+                {kv[0]: kv[1] for kv in (pair.split('=') for pair in s.split(';'))}
```

Comment:
Re kv[0]: tuple unpacked would have been more readable.

Reply:
Re `pair` and `key`. It's unfortunate that you're not sticking with the terminology coined in the spec. These name may be intuitive for you, but everyone else will have a harder time relating this code to the spec. 

I intentionally avoided the term key in the spec because its so overloaded, especially since dimension values occur as keys in the tree.

Reply:
While the spec allows multiple values on the RHS of `=`, this code does not. Is that intentional?

Reply:
Regarding comma in dimension values, each value in the dimension should produce a separate branch in the tree.